[TASK] Document options of the guides.xml (#408)
TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument: Documentation/GeneralConventions/GuidesXml.rst

    one of the main bottle-necks that can slow down rendering it can be helpful
    to limit the levels in the menu in some use cases.

..  _settings-guides-max-menu-depth-example:

Example: Limit the menu to 2 levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The changelog (documentation of the extension "core") consists of several thousands of
pages. The reason is that very long lists of changelog entries in the menu are not helpful to
the user and slow down the search.

..  code-block:: xml
    :caption: EXT:core/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
            max-menu-depth="2"
    >
        <!-- [...] -->
    </guides>

.. _settings-guides-interlink-mapping:

Interlink mapping
=================

.. todo: describe interlink mapping more detailed

The :xml:`<inventory>` tag is located directly in the :ref:`guides tag <settings-guides>`.

> There can be 0 or more :xml:`<inventory>` tags, each defines one interlink

Reviewer: ```suggestion
There can be 0 or more :xml:`<inventory>` tags, each defines one Interlink
```

Below, "Interlink" is written with a capitel "I", so this should be consistent.

    #. `<Topic> Reference`, e.g. "TSconfig Reference",
       for a complete reference
    #. `<Topic> Tutorial`, e.g. "Editors Tutorial",
       for collections of tutorials on a specific topic

.. _settings-guides-project-title-example:

Example: Set the title of a project, including version, release and copyright
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

..  code-block:: xml
    :caption: EXT:news/Documentation/guides.xml (excerpt)

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
    >
        <project title="News system" release="main (development)" version="main (development)"
                 copyright="since 2010 by Georg Ringer &amp; Contributors"/>
    </guides>

.. _settings-guides-version-and-release:

Version and release
-------------------

..  confval:: version
    :name: guides-project-version
    :type: string

    When documentation is rendered in the GitHub action for deployment onto

>     https://docs.typo3.org the version is set automatically to the extensions

Reviewer: ```suggestion
    https://docs.typo3.org the version is set automatically to the extension's
```

>     version derived from the git tag. Setting the version in the

Reviewer: ```suggestion
    version derived from the Git tag. Setting the version in the
```

    :file:`guides.xml` only affects local rendering.

    The attributes *version* and *release* both contain the version of the manual
    and mostly correspond to the version of the TYPO3 LTS or TYPO3 extension to
    which the documentation refers.

    The version is shown below the title in the theme's release switch and in the
    title meta tag, the release is not shown currently  - but it should be
    kept anyway to satisfy internal requirements.

    version
        The major project version, used as the replacement for :rst:`|version|`.
        For example this may be something like 12.4.

    release
        The full project version, used as the replacement for :rst:`|release|`

>         For example 13.0.0-dev.

Reviewer: ```suggestion
        For example 12.4.15-dev.
```

version and release should correspond, the current example does not make sense.

    a :file:`Documentation/Settings.cfg` you can use an automatic migration

>     tool to :ref:`migrate the settings.cfg into a guides.xml <migrate_guides_xml>

Reviewer 2: Here is missing closing backtick.

..  contents:: Table of Contents:
    :local:

.. _settings-guides:

The `<guides>` tag
==================

It is used for general configuration during parsing and rendering.

> The following settings can be relevant for TYPO3 themed documentation:

Reviewer: Please add that this is the root tag., so the structure is clear.

Author: done

..  _settings-guides-default-code-language:

default-code-language
---------------------

..  confval:: default-code-language
    :name: guides-default-code-language
    :type: string

    Can be used to set the code-language that should be used for
    :ref:`code-blocks <codeblock-directive>` if no language is set.

    We advice to set the language to be used in each code-block however.

..  _settings-guides-default-code-language-example:

> Example: Use `typoscript` as default langauge for code-blocks

Reviewer 2: Example: Use `typoscript` as default language for code-blocks

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

..  code-block:: xml

>     :caption: EXT:my_extension/Documentation/guides.xml (excerpt)

Reviewer: ```suggestion
    :caption: EXT:my_extension/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4
```

Emphasize the line for better reception.

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
            default-code-language="typoscript"
    >
        <!-- [...] -->
    </guides>

..  _settings-guides-max-menu-depth:

max-menu-depth
--------------

..  confval:: max-menu-depth
    :name: guides-max-menu-depth
    :type: integer
    :Default: `PHP_INT_MAX`

    Limits the main menu to a certain amount of levels. As level creation is
    one of the main bottle-necks that can slow down rendering it can be helpful

>     to the levels in the menu in some use cases.

Reviewer 2: to limit the levels in the menu in some use cases.

..  _settings-guides-max-menu-depth-example:

Example: Limit the menu to 2 levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The changelog (documentation of the extension "core") contains several thousand

Reviewer 2: The changelog (documentation of the extension "core") consists of several thousands of

> pages. As very long lists of changelog entries in the menu are not helpful to

Reviewer 2: pages. The reason is that very long lists of changelog entries in the menu are not helpful to

> the user and slow down the search

Reviewer: ```suggestion
the user and slow down the search.
```

..  code-block:: xml

>     :caption: EXT:core/Documentation/guides.xml (excerpt)

Reviewer: ```suggestion
    :caption: EXT:core/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4
```

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
            max-menu-depth="2"
    >
        <!-- [...] -->
    </guides>

.. _settings-guides-interlink-mapping:

Interlink mapping
=================

.. todo: describe interlink mapping more detailed

A list of globally available Interlink (formerly "Intersphinx") repositories
can be found in :ref:`Available default inventories <t3renderguides:available-default-inventories>`

> It is possible, though rarely needed to define custom interlink mappings:

Reviewer: Maybe add information where to embed it, something like "is a direct child of the guides tag".

Author: done

For example:

..  code-block:: xml
    :caption: Documentation/guides.xml, excerpt

    <inventory id="sphinx" url="https://www.sphinx-doc.org/en/master/"/>

.. _settings-guides-project:

The `<project>` tag
===================

> This tag can contain the following meta information:

Reviewer: Maybe add information where to embed it, something like "is a direct child of the guides tag".

Author: done

It is used for general configuration during parsing and rendering.

The following settings can be relevant for TYPO3 themed documentation:

..  _settings-guides-default-code-language:

default-code-language
---------------------

..  confval:: default-code-language
    :name: guides-default-code-language
    :type: string

    Can be used to set the code-language that should be used for
    :ref:`code-blocks <codeblock-directive>` if no language is set.

    We advice to set the language to be used in each code-block however.

..  _settings-guides-default-code-language-example:

> Example: Use `typoscript` as default langauge for code-blocks

Reviewer 2: ```suggestion
Example: Use `typoscript` as default language for code-blocks
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

..  code-block:: xml
    :caption: EXT:my_extension/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
            default-code-language="typoscript"
    >
        <!-- [...] -->
    </guides>

..  _settings-guides-max-menu-depth:

max-menu-depth
--------------

..  confval:: max-menu-depth
    :name: guides-max-menu-depth
    :type: integer
    :Default: `PHP_INT_MAX`

    Limits the main menu to a certain amount of levels. As level creation is
    one of the main bottle-necks that can slow down rendering it can be helpful

>     to the levels in the menu in some use cases.

Reviewer 2: ```suggestion
    to limit the levels in the menu in some use cases.
```

..  _settings-guides-max-menu-depth-example:

Example: Limit the menu to 2 levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> The changelog (documentation of the extension "core") contains several thousand

Reviewer 2: ```suggestion
The changelog (documentation of the extension "core") consists of several thousands of
```

> pages. As very long lists of changelog entries in the menu are not helpful to

Reviewer 2: ```suggestion
pages. The reason is that very long lists of changelog entries in the menu are not helpful to
```

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
    >
        <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
                   edit-on-github-branch="main"
                   edit-on-github="typo3/typo3"
                   edit-on-github-directory="typo3/sysext/adminpanel/Documentation/"
        />
    </guides>

.. _settings-guides-interlink-shortcode:

interlink-shortcode
-------------------

..  confval:: interlink-shortcode
    :name: guides-extension-interlink-shortcode
    :type: string
    :Default: `somemanual`

    The interlink shortcode will be displayed in the "Reference this headline"
    dialog when users click on the link symbol beside a headline or other
    linkable element.

    These references can be used to link to your manual from other documents
    rendered with the TYPO3 theme. However they will only work if you follow
    these rules:

>     *   TYPO3 third party extensions: composer name, example: `georgringer/news`

Reviewer: ```suggestion
    *   TYPO3 third-party extensions: Composer name, example: `georgringer/news`
```

>     *   System extensions: composer name, example: `typo3/cms-adminpanel`

Reviewer: ```suggestion
    *   System extensions: Composer name, example: `typo3/cms-adminpanel`
```

    *  `https://typo3.slack.com/archives/C028JEPJL`.

.. _settings-guides-project_repository:

project_repository
~~~~~~~~~~~~~~~~~~

..  confval:: project_repository
    :name: guides-extension-project_repository
    :type: string

    is set to the repository of the project's VCS, for
    example

    *  `https://github.com/FriendsOfTYPO3/extension_builder`.

.. _settings-guides-project_issues:

project_issues
~~~~~~~~~~~~~~

..  confval:: project_issues
    :name: guides-extension-project-issues
    :type: string

    is set to the location where project issues are to be
    created and edited, for example

    *  `https://github.com/FriendsOfTYPO3/extension_builder/issues`

.. _settings-guides-project_discussions:

project_discussions
~~~~~~~~~~~~~~~~~~~

..  confval:: project_discussions
    :name: guides-extension-project-discussions
    :type: string

    is used in the rare case that project-related
    discussions take place in locations other than those defined by the
    project_contact and project_issues attributes, for example

    *  `https://github.com/FriendsOfTYPO3/extension_builder/discussions`.

.. _settings-guides-project-links-example:

> Example: Project links for a third party extension

Reviewer: ```suggestion
Example: Project links for a third-party extension
```

            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
         >
        <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
                   project-home="https://docs.typo3.org/m/typo3/reference-tsconfig/main/en-us/"
                   project-contact="https://typo3.slack.com/archives/C028JEPJL"
                   project-repository="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-TSconfig"
                   project-issues="https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-TSconfig/issues"
        />
    </guides>

..  _settings-guides-preferred_typo3_version:

typo3-core-preferred
--------------------

..  confval:: typo3-core-preferred
    :name: guides-extension-typo3-core-preferred
    :type: string
    :Default: `stable`

    You can set the preferred TYPO3 Core version. This version is used to determine
    the desired version for links to other manuals and system extensions.

    ..  code-block:: xml
        :caption: Documentation/guides.xml

        <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"

>                    typo3-core-preferred="12.4" />

Reviewer: ```suggestion
                   typo3-core-preferred="stable" />
```

I suggest to use "stable" as people tend to copy/paste those and will come up later with references to outdated manuals (when not adapted manually).

and v12.4. When :ref:`referencing <cheat-sheet-intersphinx>` Core manuals or
extensions, we want the links to go automatically to version 11.5 of that
reference or core extension. Therefore we set the preferred TYPO3 Core version
to that value:

..  code-block:: xml
    :caption: EXT:news/Documentation/guides.xml (excerpt)

    <?xml version="1.0" encoding="UTF-8"?>
    <guides xmlns="https://www.phpdoc.org/guides" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="https://www.phpdoc.org/guides ../vendor/phpdocumentor/guides-cli/resources/schema/guides.xsd"
    >
         <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
               typo3-core-preferred="11.5"
    </guides>

Example: Always link to the latest stable TYPO3 version
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you want your links to always go to the latest stable long time support version
(LTS) of the TYPO3 Core, you can set :confval:`guides-extension-typo3-core-preferred` to stable.

At the time of writing references to official manuals and Core extensions will

> be rendered to version 12.4. Once TYPO3 13LTS is released they will automatically

Reviewer: ```suggestion
be rendered to version 12.4. Once TYPO3 13 LTS is released they will automatically
```
